homeassistant/components/airvisual/__init__.py

```diff
@@ -206,6 +206,11 @@ def _standardize_geography_config_entry(
 
 async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
     """Set up AirVisual as config entry."""
+    if CONF_API_KEY not in entry.data:
+        # If this is a migrated AirVisual Pro entry, there's no actual setup to do;
+        # that will be handled by the `airvisual_pro` domain:
+        return True
```

Comment:
It's weird that we let listeners know that the config entry is setup but it's actually been removed. Not sure if this is ok.

I'd consider scheduling the remove instead, so it doesn't happen before we return here.

Reply:
As discussed on Discord, I'm still trying to figure out how to schedule such a removal. I've tried listening to both `EVENT_HOMEASSISTANT_STARTED` and `EVENT_COMPONENT_LOADED,` but in both cases, I get this error:

```
2022-12-29 13:49:50.632 ERROR (MainThread) [homeassistant.config_entries] Error unloading entry Node/Pro (office-airvisual-node.phil.iot) for sensor
Traceback (most recent call last):
  File "/Users/bachya/Git/home-assistant/homeassistant/config_entries.py", line 533, in async_unload
    result = await component.async_unload_entry(hass, self)
  File "/Users/bachya/Git/home-assistant/homeassistant/components/sensor/__init__.py", line 575, in async_unload_entry
    return await component.async_unload_entry(entry)
  File "/Users/bachya/Git/home-assistant/homeassistant/helpers/entity_component.py", line 181, in async_unload_entry
    raise ValueError("Config entry was never loaded!")
ValueError: Config entry was never loaded!
```

It seems strange to have `airvisual_pro` remove the old config entry, as it would have to wait until this old config entry is fully loaded—that feels too intertwined.